### PR TITLE
Refactor FlagsWidget to use QAbstractItemModel (and some other details)

### PR DESCRIPTION
**Pull request by thestr4ng3r**
Basically the same what I did with the FunctionsWidget, but much simpler since the FlagsWidget does not have the nested view. I also added the Quick Filter field.

Additionally, I removed the default value for the -A option, because this didn't work the way I thought. Unfortunately, QCommandLineParser does not support omitting the value for an option like this.

I have also moved the pkg-config stuff in CMake to FindRadare2.cmake to keep the actual CMakeLists.txt clean.

### CODE REVIEW

File: src/widgets/flagswidget.cpp [hunk 1]
```diff
@@ -9,14 +9,138 @@
...
 
+
+FlagsModel::FlagsModel(QList<FlagDescription> *flags, MainWindow *main, QObject *parent)
+    : QAbstractListModel(parent),
+      main(main),
+      flags(flags)
+{
+}
+
+int FlagsModel::rowCount(const QModelIndex &parent) const
```
Comment (ballessay) [inline src/widgets/flagswidget.cpp]: warning: unused parameter ‘parent’ -> don't name the parameter

File: src/widgets/flagswidget.cpp [hunk 2]
```diff
@@ -9,14 +9,138 @@
...
+{
+    return flags->count();
+}
+
+int FlagsModel::columnCount(const QModelIndex &parent) const
```
Comment (ballessay) [inline src/widgets/flagswidget.cpp]: warning: unused parameter ‘parent’ . see rowCount

File: src/widgets/flagswidget.h [hunk 3]
```diff
@@ -1,11 +1,58 @@
...
+
+
+
+
+class FlagsModel: public QAbstractListModel
+{
+    Q_OBJECT
+
+private:
+    MainWindow *main;
```
Comment (ballessay) [inline src/widgets/flagswidget.h]: `main` is not used in the model -> remove? future use?

**Review by ballessay — COMMENTED** [review]
Minor stuff. Overall looks good to me.

**Comment by thestr4ng3r** [comment]
Done. Thanks for the feedback!

**Pull request merged**